### ISSUE TITLE
[kompile] Add option for post-processing the kore output

### ISSUE THREAD

**Issue by ehildenb** [runtimeverification/k#2770]
Currently we have `--post-process`, which happens after Kompile pipeline, and before ModuleToKORE.

pyk now supports Kore data-structures directly, and parsing/unparsing of them. So we can also easily have `--post-process-kore` option, which passes the Kore through an external program and then reads it back in. This would enable things like this: https://github.com/runtimeverification/k/pull/2768, where we want to mangle the definition a bit before passing it to the backend, and would also make things like the Dedukti backend easier.

We can rename `--post-process` to `--post-process-kast`, and then add option `--post-process-kore` for this use-case.

**Comment by radumereuta** [comment]
But do we need a `-post-process-kore` option?
You just call a script after kompile finishes that modifies the `definition.kore` file.

**Comment by ehildenb** [comment]
True, would be for smoother integration I suppose. Also because after the definition is generated, then the haskell backend validation routines are called on it. So we probably want to validate the post-processed definition, not the pre-processed version, to double-check that the user isn't doing anything that breaks assumptions about Kore.
